common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -84,12 +81,6 @@ type (
 
 	visibilityPageToken struct {
 		SearchAfter []interface{}
-
-		// For ScanWorkflowExecutions API.
-		// For ES<7.10.0 and "oss" flavor.
-		ScrollID string
-		// For ES>=7.10.0 and "default" flavor.
-		PointInTimeID string
```

Comment:
This doesn't look backwards-compatible because we just [parse this message and return an error if that didn't work](https://github.com/temporalio/temporal/blob/0d6e92f8b300fdbd6bd9cf46f91d35fdbad32d3d/common/persistence/visibility/store/elasticsearch/visibility_store.go#L820). Could you please add a test case which verifies that old page tokens will still be handled gracefully?

Reply:
Decoding old token won't throw any error. The query will work and the only difference is that we won't set the scroll id nor the PIT id. So, the result might be unexpected, but it won't fail.

Reply:
I'm a little skeptical about that. Could you add a backwards compatibility test? Something like
`s.NoError(Serialize(oldDef).Deserialize(&newDef))`

Reply:
Added a test calling Scan API with old token.